### DISCUSSION TITLE
Add beatmap offset hotkeys into lazer

### DISCUSSION THREAD

**Discussion by frarg** [ppy/osu#21882] [Ideas]
I think this feature is fairly simple but (as far as I know) is missing in lazer. 

Simply adding the ability to press + or - to apply beatmap offsets in increments of 5ms would be really nice for the muscle memory built up from stable (I am someone who uses local offsets A LOT).

Also perhaps adding the ability to change offset while the map is paused if that's possible now (I know it wasn't possible in stable)

**Comment by bdach** [comment]
> Also perhaps adding the ability to change offset while the map is paused if that's possible now (I know it wasn't possible in stable)

That sounds questionable from a game balance standpoint.

**Reply by frarg** [reply]
I hadn't actually thought about it like that. I can't really imagine a scenario where someone could realistically gain an advantage with this anyways but I can understand the concern.

My main point was just for convenience sake: instead of having to rush to change it at the start of a map, being able to change it while paused and having it only apply once you retry would be fine.

**Reply by Inconsist** [reply]
The simplest way to avoid it really is to calculate the time from the last hitwindow and only allow "paused change" up to that time. Avoids the"hitting objects that are in the past" problem while still allowing changes during short breaks. Seems like the best workaround to me.

**Comment by givowo** [comment]
the -/+ hotkeys not being in lazer is the main reason why i wont play lazer. I change offset so often that its almost muscle memory

**Reply by Inconsist** [reply]
Try to use the quick retry adjust offset button on the right. It calculates it for you and I find it works pretty well. Though a hotkey to apply automated offet would be neat...

**Reply by Axilotl17** [reply]
however in a multiplayer lobby that doesn't work.

**Reply by givowo** [reply]
> Try to use the quick retry adjust offset button on the right. It calculates it for you and I find it works pretty well. Though a hotkey to apply automated offet would be neat...

well ok but i would rather do it manually

**Comment by peppy** [comment]
If anyone is looking to implement this, it should be limited to the first x seconds of the beatmap unless in break time, similar to stable. It should also be rate limited to avoid any kind of abuse.

stable implementation:

```csharp
        private void ChangeCustomOffset(int msChange)
        {
            if (Paused || Unpausing)
            {
                NotificationManager.ShowMessageMassive(LocalisationManager.GetString(OsuString.Player_OffsetChangeUnpause), 1000);
                return;
            }

            if (AudioEngine.Time > firstHitTime + 10000 && !GameBase.TestMode && !EventManager.BreakMode)
                return;

            if (GameBase.Time - lastPlayerOffsetChange < 150)
                return;
        }

```

---

A better approach may be to make the offset calibration control usable during breaks?

**Reply by Inconsist** [reply]
As already said in #22214 it would alos be nice to be able to open this nameless menu at the beginning of a map with a keybind so you dont have to restart a map everytime your cursor accidentally leaves the menu.
